### PR TITLE
Include test frameworks

### PR DESCRIPTION
**Pull request by akenzua**
This particular PR adds test framework to a react prototype that is supposed to be a boilerplate for all other directory projects that will be done in react. We were working one that did not factor in non js users. This is an update on [react_prototype](https://github.com/moneyadviceservice/rails_prototype) it would have been a branch of it but because it is structurally different it  has to be made into another repo. 
It is meant to be without pages and  styles but I left all the styles and components in there in anticipation of the react library being worked on by @andreferreiradlw because it can not be entirely empty. Stuffs like header and footer should readily present at least

### CODE REVIEW

**Comment by davidtrussler** [comment]
I have a couple of minor comments that are more pertinent to the master branch: 
- Could we add: `cd isomorphic_react_prototype` in the installation section, before `npm install` please?
- do we need a `.gitignore` file? I get a shedload of created and changed files when I install this that I'm guessing I don't need.

**Comment by akenzua** [comment]
Thank @davidtrussler i have correct that part of the read me.

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        "ecmaVersion": 2018,
+        "sourceType": "module"
+    },
+    "plugins": [
+        "react"
+    ],
+    "rules": {
+        "react/react-in-jsx-scope": "off"
+    }
+}
```
Comment (andreferreiradlw) [inline .eslintrc.json]: Can you please add some plugins to eslint to make it more consistent across apps? Please follow my config from [here](https://github.com/moneyadviceservice/react_library/blob/master/.eslintrc)

File: package.json [hunk 2]
```diff
@@ -31,6 +33,19 @@
...
+    "babel-jest": "^24.9.0",
+    "babel-plugin-styled-components": "^1.10.6",
+    "enzyme": "^3.11.0",
+    "enzyme-adapter-react-16": "^1.15.2",
+    "eslint": "^6.8.0",
+    "eslint-config-airbnb": "^18.0.1",
+    "eslint-plugin-import": "^2.20.0",
+    "eslint-plugin-jsx-a11y": "^6.2.3",
+    "eslint-plugin-react": "^7.17.0",
+    "eslint-plugin-react-hooks": "^2.3.0",
```
Comment (andreferreiradlw) [inline package.json]: Refering to my previous comment, you're importing all the plugins for eslint but not using them

**Comment by andreferreiradlw** [comment]
Also the express server is not starting for me. I'm getting this:

`
{ [Error: ENOENT: no such file or directory, open '/Users/andre/Repos/isomorphic_react_prototype/.next/cache/autodll-webpack-plugin/development_instance_0_cd1f809de12695cef398f784821523c6/stats.json']
  cause:
   { Error: ENOENT: no such file or directory, open '/Users/andre/Repos/isomorphic_react_prototype/.next/cache/autodll-webpack-plugin/development_instance_0_cd1f809de12695cef398f784821523c6/stats.json'
     errno: -2,
     code: 'ENOENT',
     syscall: 'open',
     path:
      '/Users/andre/Repos/isomorphic_react_prototype/.next/cache/autodll-webpack-plugin/development_instance_0_cd1f809de12695cef398f784821523c6/stats.json' },
  isOperational: true,
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path:
   '/Users/andre/Repos/isomorphic_react_prototype/.next/cache/autodll-webpack-plugin/development_instance_0_cd1f809de12695cef398f784821523c6/stats.json' }
`

**Comment by andreferreiradlw** [comment]
I think you should add the cache file for webpack to the `.gitignore` file as well

**Comment by akenzua** [comment]
I have update the eslint config as necessary. However, this project is  nextjs and [react_library](https://github.com/moneyadviceservice/react_library) is a pure react project (am assuming); their linting requirements are not the same. Somethings are taken care of out of the box and some configs are hidden. For example, you do not import react nextjs, it is not necessary.
Moreso, as we use the prototype for other projects, there may be need to update the eslint config as we may adopt more libraries and tools not in the original prototype.

**Comment by akenzua** [comment]
The .next folder has been added to gitignore file and all associated problem has been resolved.

**Pull request merged**